connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaExecutable.java

```diff
@@ -0,0 +1,234 @@
+
+  Consumer<String, String> consumer;
+
+  private final Function<Properties, Consumer<String, String>> consumerCreatorFunction;
+
+  public CompletableFuture<?> future;
+
+  private InboundConnectorContext context;
+
+  boolean shouldLoop = true;
+
+  public KafkaExecutable(
+      final Function<Properties, Consumer<String, String>> consumerCreatorFunction) {
+    this.consumerCreatorFunction = consumerCreatorFunction;
+  }
+
+  public KafkaExecutable() {
+    this(KafkaConsumer::new);
+  }
+
+  @Override
+  public void activate(InboundConnectorContext connectorContext) {
+    this.context = connectorContext;
+    KafkaConnectorProperties elementProps =
+        connectorContext.getPropertiesAsType(KafkaConnectorProperties.class);
+    connectorContext.replaceSecrets(elementProps);
+    connectorContext.validate(elementProps);
+    this.consumer =
+        createConsumer(
+            this.consumerCreatorFunction,
+            getKafkaProperties(elementProps, this.context),
+            elementProps,
+            getOffsets(elementProps.getOffsets()));
+    LOG.debug("Kafka inbound connector initialized");
+    this.future =
+        CompletableFuture.supplyAsync(
+            () -> {
+              try {
+                while (shouldLoop) {
+                  ConsumerRecords<String, String> records =
+                      this.consumer.poll(Duration.ofMillis(500));
+                  for (ConsumerRecord<String, String> record : records) {
+                    LOG.trace(
+                        "Kafka message received: key = %s, value = %s%n",
+                        record.key(), record.value());
+                    InboundConnectorResult<?> result =
+                        connectorContext.correlate(
+                            convertConsumerRecordToKafkaInboundMessage(record));
+                    if (result.isActivated()) {
+                      LOG.debug(
+                          "Inbound event correlated successfully: {}", result.getResponseData());
+                    } else {
+                      LOG.debug("Inbound event not correlated: {}", result.getErrorData());
+                    }
+                  }
+                  if (!records.isEmpty()) {
+                    this.consumer.commitSync();
+                  }
+                }
+              } catch (ConnectorInputException e) {
+                LOG.warn("Failed to parse message body: {}", e.getMessage());
+                throw e;
+              } catch (Exception ex) {
+                LOG.error("Failed to execute connector: {}", ex.getMessage());
+                throw ex;
+              }
+              LOG.debug("Kafka inbound loop finished");
+              return null;
+            });
+  }
+
+  @Override
+  public void deactivate() {
+    try {
+      this.shouldLoop = false;
+      if (this.future != null && !this.future.isDone()) {
+        this.future.get();
+      }
+      this.shouldLoop = true;
```

Comment:
Although it was working as expected, you are right, line 127 is not necessary, since for every activate call we should always create a new instance of the class with the shouldLoop initialized as true. I removed this line.